pkg/events/event_store.go

```diff
@@ -68,8 +68,12 @@ func (es *EventStore) CollectEvents() []*si.EventRecord {
 	defer es.Unlock()
 
 	messages := es.events[:es.idx]
-	es.events = make([]*si.EventRecord, defaultEventStoreSize)
+	if es.size != es.lastSize {
+		log.Log(log.Events).Info("Resizing event store", zap.Uint64("last", es.lastSize), zap.Uint64("new", es.size))
+	}
+	es.events = make([]*si.EventRecord, es.size)
```

Comment:
Do we need to reallocate array if the `idx` can be used to guard against incorrect reading.

Reply:
correct, it's not needed, new code does not reallocate slice storage every time